src/ring_theory/finiteness.lean

```diff
@@ -260,10 +274,30 @@ end
 /-- If `f : A →ₐ[R] B` is surjective with finitely generated kernel and `A` is finitely presented,
 then so is `B`. -/
 lemma of_surjective {f : A →ₐ[R] B} (hf : function.surjective f) (hker : f.to_ring_hom.ker.fg)
-  (hfp : finitely_presented R A) : finitely_presented R B :=
+  (hfp : finite_presentation R A) : finite_presentation R B :=
 equiv (quotient hker hfp) (ideal.quotient_ker_alg_equiv_of_surjective hf)
 
-end finitely_presented
+/-- An algebra is finitely presented if and only if it is a quotient of a polynomial ring whose
+variables are indexed by a fintype by a finitely generated ideal. -/
+lemma iff_quotient_mv_polynomial : finite_presentation R A ↔ ∃ (ι : Type) [fintype ι]
```

Comment:
I tried to have here `∃ (ι : Type*)` but I didn't managed to do it (to be honest I don't even understand if this is true or not...).

Reply:
Perhaps worth calling this `iff_quotient_mv_polynomial'` to match the prime-ness of the lemma about `finite_type`

Reply:
Ah, you're right.

I'm wondering if the version with `finset` is worth to be proven. I wrote the one for `fintype`, but I now have the impression that it is a quite unnatural statement, since the type of elements of the finset must be the big ring, so it is the version with fintype that really models what we (or at least I) have in mind for a finitely generated algebra.